src/components/AppToolbar/RoleSelection.tsx

```diff
@@ -35,7 +35,12 @@ const RoleSelection: React.FC<RoleSelectionProps> = ({ close }) => {
       }
     };
     getUserInformation();
-  }, [user.id, api]);
+
+    return () => {
+      unmounted = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I wonder does passing in [user.id, api] cause any issues?

Reply:
Yes it was causing because when we update the token user is updated as well in the user context and it was trying to execute useEffect again but that component is already removed because we re-render the view. It was firing error or warning in the console that `Can't perform a React state update on an unmounted component`.